### ISSUE TITLE
Input array shape inconsistency

### ISSUE THREAD

**Issue by ixjlyons** [kb-press/ndsplines#35]
In writing some API coverage tests (I/O array shapes), I found the following:

```python
import ndsplines
import numpy as np

# case 1: x.ndim == 1, y.ndim == 1
# can sample with xx.ndim == 1
x = np.linspace(0, 1, 10)
xx = np.linspace(0, 1, 100)
y = np.sin(x)
spl = ndsplines.make_interp_spline(x, y)
yy = spl(xx)

# case 2: x.ndim == 1, y.ndim == 2
# have to have xx shape (1, n)
y = np.stack((np.sin(x), np.cos(x)))
spl = ndsplines.make_interp_spline(x, y)
yy = spl(xx[None, :])
yy = spl(xx) # fails
```

The last line generates:

```
Traceback (most recent call last):
  File "1d-cases.py", line 18, in <module>
    yy = spl(xx) # fails
  File "/home/kenny/src/ndsplines/ndsplines/ndsplines.py", line 165, in __call__
    return y_out.reshape((self.ydim,) + x_shape[1:])
ValueError: cannot reshape array of size 200 into shape (2,)
```

We don't necessarily have to handle `xx.shape == (n,)`, but it's a little inconsistent. I think we could just add `x = np.atleast_2d(x)` or equivalent (`if x.ndim == 1: x = x[None, :]`) in `__call__` but not sure yet if that has any side effects.

**Comment by ixjlyons** [comment]
Another note that may be related:

```python
import ndsplines
import numpy as np

x = np.linspace(0, 1, 10)
y = np.sin(x)
spl = ndsplines.make_interp_spline(x, y)
print(spl.ydim) # -> 10

x = np.linspace(0, 1, 10)[None, :]
y = np.sin(x)
spl = ndsplines.make_interp_spline(x, y)
print(spl.ydim) # -> 1
```

**Comment by sixpearls** [comment]
Glad you caught this! 

I had already seen the problem you identified in the first comment for the "squeeze"d case, and didn't propagate that for the non-squeezed case.

The second one is a missed update that I'm going to blame on juggling branches. This [line](https://github.com/kb-press/ndsplines/blob/d2330486d636100dfce472d15e0b3deea928702b/ndsplines/ndsplines.py#L56) should have been updated to reference `self.coefficients`.

Both are fixed by #36. I'll leave it to you to merge when it is most convenient for your test writing.
